src/interpreter/builder.rs

```diff
@@ -68,20 +69,31 @@ where
         })
     }
 
-    pub fn build(mut self) -> Fallible<Interpreter<'a, Op>> {
+    pub fn build(self) -> Fallible<Interpreter<'a, Op>> {
+        self.build_with_threads(-1)
+    }
+
+    pub fn build_with_threads(mut self, threads: c_int) -> Fallible<Interpreter<'a, Op>> {
```

Comment:
How about annotating this and `set_num_threads` with `#[cfg(feature = "multi_thread")]`?

Reply:
I started out doing that then realized I'd have to duplicate logic in the builder and left it ungated because the functions are always there but they just don't do anything. I'll make the change and push it in a bit